contrib/static-web-server/StaticWebServer.hs

```diff
@@ -15,13 +16,7 @@ main = serverWith defaultConfig { srvHost = "0.0.0.0", srvLog = stdLogger, srvPo
     GET -> do
       let ext = takeExtension (url_path url)
       putStrLn $ url_path url
-      mb_txt <- try (readFile $ url_path url)
-      case mb_txt of
-        Right a -> return $ sendText ext OK a
-        Left e -> return $ sendText ".html" NotFound "Not found"
-
-                 where _hack :: SomeException
-                       _hack = e   -- to specify the type
+      returnFileContent ext url `catch` (\(_ :: IOException) -> return $ sendText ".html" NotFound "Not Found")
```

Comment:
Either use 
- [System.IO.Error](http://hackage.haskell.org/package/base-4.8.2.0/docs/System-IO-Error.html) to detect "not found":

`f = catchIOError g (\e -> if IO.isEOFError e then return [] else ioError e)`

- Or `fileExists` from [System.Directory](http://hackage.haskell.org/package/directory-1.2.5.1/docs/System-Directory.html)